src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,63 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import cn from 'classnames';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+import { getNormalizeMovie } from '../../utils/getNormalizeMovie';
+
+type Props = {
+  addMovie: (movie: Movie) => void
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSearch = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    try {
+      const movieFromServer = await getMovie(query);
+
+      if ('Error' in movieFromServer) {
+        setHasError(true);
+        setIsLoading(false);
+
+        return;
```

Comment:
nice case, to put to `catch`

Reply:
I can't put it inside catch, because typescript will show an error below when I setMovie and its possible error object.